graph/src/components/link_resolver.rs

```diff
@@ -53,19 +77,158 @@ impl LinkResolver for ipfs_api::IpfsClient {
             None => Box::new(cat),
         }
     }
+
+    fn json_stream(
+        &self,
+        link: &Link,
+    ) -> Box<Future<Item = ValueStream, Error = failure::Error> + Send + 'static> {
+        // Discard the `/ipfs/` prefix (if present) to get the hash.
+        let path = link.link.trim_start_matches("/ipfs/").to_owned();
+        let mut stream = self.cat(&path).fuse();
+        let mut buf = BytesMut::with_capacity(1024);
```

Comment:
I _think_ that's ok, because we are going to create a bunch of objects in memory that represent the entire file anyway, hoping it will be some constant times `GRAPH_MAX_IPFS_MAP_FILE_SIZE`, so adding 1 to that constant doesn't strike me as that bad, especially since a file with one line that's a few hundred MB seems pretty unlikely.